### PR TITLE
Make embeds work for all server-side rendered samples

### PR DESCRIPTION
**Pull request by sebastianbenz**
* introduce backend sample API
* refactor existing samples to use the new API

### CODE REVIEW

**Comment by sebastianbenz** [comment]
@kul3r4 PTAL

**Comment by sebastianbenz** [comment]
Fixed the build failure

File: backend/comment-section.go [hunk 1]
```diff
@@ -23,11 +23,9 @@ import (
 )
 
 const (
-	COMMENT_SAMPLE_PATH         = "/samples_templates/comment_section/"
-	COMMENT_SAMPLE_PATH_PREVIEW = COMMENT_SAMPLE_PATH + "preview/"
-	COMMENT_COOKIE_NAME         = "ABE_LOGGED_IN"
-	SUBMIT_COMMENT_XHR          = "submit-comment-xhr"
-	USER                        = "Charlie"
+	COMMENT_SAMPLE_PATH = "/" + CATEGORY_SAMPLE_TEMPLATES + "/comment_section/"
```
Comment (kul3r4) [inline backend/comment-section.go]: maybe CATEGORY_SAMPLE_TEMPLATES -> CATEGORY_SAMPLES_TEMPLATES given the original value contains the s?

Comment (sebastianbenz) [inline backend/comment-section.go]: done

**Comment by kul3r4** [comment]
LGTM

**Comment by sebastianbenz** [comment]
I haven't pushed my changes yet.

**Pull request merged**